### PR TITLE
bpf: Use upper 16 bits of mark for identity

### PR DESCRIPTION
**Pull request by tgraf**
The skb->mark is being used by Kubernetes to indicate packets which require
masquerading or packets that should be dropped.

The bit overlapped with the identity space which had to potential for
undesirable side effects.

Keep the 4 bits unused which are being utilized by Kubernetes and instead use
the upper 16 bits for the identity and the unused lower 12 bits for the magic
pattern.

Signed-off-by: Thomas Graf <thomas@cilium.io>

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Review by aanm — APPROVED** [review]
If I upgrade cilium with these change there will be any problems for running endpoints? (I'm 99% sure there isn't, but just trying to make sure)

**Comment by tgraf** [comment]
> If I upgrade cilium with these change there will be any problems for running endpoints? (I'm 99% sure there isn't, but just trying to make sure)

There will be no problem because:
 - All ip rules will get replaced with new rules without exact string matches on old rules
 - The agent and the proxies are tied to each other so Envoy and the Kafka proxy receive this change at the same time.
 - When endpoints get restored, they will receive the new pattern on regeneration.

**Review by borkmann — APPROVED** [review]
lgtm

**Comment by tgraf** [comment]
Kubernetes Test flake:
```
00:06:32 Unable to connect to the server: net/http: TLS handshake timeout
```

**Comment by ianvernon** [comment]
@tgraf that flake happens periodically, most likely a `kubeadm` bug :(

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: pkg/proxy/proxy.go [hunk 1]
```diff
@@ -69,12 +69,20 @@ type Redirect interface {
 // GetMagicMark returns the magic marker with which each packet must be marked.
 // The mark is different depending on whether the proxy is injected at ingress
 // or egress.
-func GetMagicMark(isIngress bool) int {
+func GetMagicMark(isIngress bool, identity int) int {
```
Comment (raybejjani) [inline pkg/proxy/proxy.go:75]: Should this return a uint? perhaps even a uint32?

**Review by raybejjani — APPROVED** [review]
Can we add a define or something to reserve the bit k8s uses? I doubt we'll touch this again but it might help once we finally do come back.

File: pkg/proxy/proxy.go (hunk 1 above)
Comment (tgraf) [reply pkg/proxy/proxy.go:75]: Changing to uint32 will be a bigger change. We should not be losing any bits anywhere.

**Comment by tgraf** [comment]
> Can we add a define or something to reserve the bit k8s uses? I doubt we'll touch this again but it might help once we finally do come back.

Fixed

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**